config/initializers/active_record.rb

```diff
@@ -40,36 +40,28 @@ def to_json(options = {})
 
     def self.conditions_for_date(column, date)
       if date == 'today'
-        return ["#{column}::DATE = ?", Date.today.to_s]
+        return ["#{column} >= ? AND #{column} < ?", Date.today.to_s, Date.tomorrow.to_s]
       end
       year, month, day = date.to_s.split('-').map do |d|
         d = d.blank? ? nil : d.to_i
         d == 0 ? nil : d
       end
-      if date.to_s =~ /^\d{4}/ && year && month && day
-        datestring = "#{year}-#{month}-#{day}"
+      if year || month || day
         begin
-          Date.parse(datestring)
-          ["#{column}::DATE = ?", datestring]
+          if day
+            date = Date.parse("#{ year }-#{ month }-#{ day }")
+            [ "#{column} >= ? AND #{column} < ?", date, date + 1.day ]
+          elsif month
+            date = Date.parse("#{ year }-#{ month }-1")
+            [ "#{column} >= ? AND #{column} < ?", date, date + 1.month ]
```

Comment:
Does this replicates the former functionality, where requesting `/observations?month=6` would show observations observed in June of *any* year?